include/vctr/Expressions/BasicMath/NormalizeSum.h

```diff
@@ -0,0 +1,81 @@
+/*
+  ==============================================================================
+    DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+
+    Copyright 2023 by sonible GmbH.
+
+    This file is part of VCTR - Versatile Container Templates Reconceptualized.
+
+    VCTR is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License version 3
+    only, as published by the Free Software Foundation.
+
+    VCTR is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License version 3 for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    version 3 along with VCTR.  If not, see <https://www.gnu.org/licenses/>.
+  ==============================================================================
+*/
+
+namespace vctr::expressions
+{
+
+template <size_t extent, class SrcType>
+requires is::floatNumber<ValueType<SrcType>>
+class NormalizeSum : ExpressionTemplateBase
+{
+public:
+    using value_type = ValueType<SrcType>;
+
+    VCTR_COMMON_UNARY_EXPRESSION_MEMBERS (NormalizeSum, src)
+
+    void applyRuntimeArgs()
```

Comment:
`applyRuntimeArgs()` is called as soon as a source container is prepended to a (chain of) expression(s). Before, it has been used to pass back runtime values like e.g. the bounds for the `clamp` expression or the range for the `map` expression. I figured out that in this particular case we need the option to compute the sum once before each call to `operator[]` or the other evaluation functions, so I also made it possible that this function can be used without any arguments.

Still you got a point here. There is one particular edge case here which is this example:
```c++
vctr::Vector v { 1.0, 2.0, 3.0 };

auto normalizedSum = vctr::normalizeSum << v;

vctr::Vector res1 = normalizedSum;

v.push_back (4.0);
v[2] = 5.0;

// At this point the normalizedSum expression will still hold the sum of the elements of v from the time point
// where the expression has been created. It will not re-evaluate it here because it does not assume that the
// source values have changed. 
vctr::Vector res2 = normalizedSum;
```

The pattern sketched above where an unevaluated expression is stored to a variable and then is evaluated multiple times while changing values of the source container(s) should just work with all other expressions so far. Although this is a relatively uncommon use-case I think it is actually a flaw of this particular expression that is hard to avoid if we don't want to re-evaluate the sum on each invocation of operator[] which would be a nightmare performance wise.

Tbh. I'm not sure if this expression is more harmful than useful, given that code using it could easilly be rewritten using a straightforward sum and division. We could keep it and add a big note regarding that, we could keep it and declare it deprecated from the start on just to make it easier to rework old code relying on the old `normalizeSum` implementation or we could just not add it and just rework current internal code using `normalizeSum` when rolling out VCTR in our internal code base. What do you think?
I added it